numpy/polynomial/polynomial.py

```diff
@@ -780,6 +781,95 @@ def polyval(x, c, tensor=True):
     return c0
 
 
+def polyvalfromroots(x, r, tensor=True):
+    """
+    Evaluate a polynomial specified by its roots at points x.
+
+    If `r` is of length `N`, this function returns the value
+
+    .. math:: p(x) = \prod_{n=1}^{N} (x - r_n)
+
+    The parameter `x` is converted to an array only if it is a tuple or a
+    list, otherwise it is treated as a scalar. In either case, either `x`
+    or its elements must support multiplication and addition both with
+    themselves and with the elements of `r`.
+
+    If `r` is a 1-D array, then `p(x)` will have the same shape as `x`.  If
+    `r` is multidimensional, then the shape of the result depends on the
+    value of `tensor`. If `tensor` is true the shape will be r.shape[1:] +
+    x.shape. If `tensor` is false the shape will be r.shape[1:]. Note that
+    scalars have shape (,).
+
+    Parameters
+    ----------
+    x : array_like, compatible object
+        If `x` is a list or tuple, it is converted to an ndarray, otherwise
+        it is left unchanged and treated as a scalar. In either case, `x`
+        or its elements must support addition and multiplication with
+        with themselves and with the elements of `r`.
+    r : array_like
+        Array of roots. If `r` is multidimensional the remaining indices
+        enumerate multiple polynomials. In the two dimensional case the
+        coefficients may be thought of as stored in the columns of `r`.
+    tensor : boolean, optional
+        If True, the shape of the coefficient array is extended with ones
+        on the right, one for each dimension of `x`. Scalars have dimension 0
+        for this action. The result is that every column of coefficients in
+        `r` is evaluated for every element of `x`. If False, `x` is broadcast
+        over the columns of `r` for the evaluation.  This keyword is useful
+        when `r` is multidimensional. The default value is True.
+
+        .. versionadded:: 1.12
+
+    Returns
+    -------
+    values : ndarray, compatible object
+        The shape of the returned array is described above.
+
+    See Also
+    --------
+    polyroots, polyfromroots, polyval
+
+    Examples
+    --------
+    >>> from numpy.polynomial.polynomial import polyvalfromroots
+    >>> polyvalfromroots(1, [1,2,3])
+    0.0
+    >>> a = np.arange(4).reshape(2,2)
+    >>> a
+    array([[0, 1],
+           [2, 3]])
+    >>> polyvalfromroots(a, [-1, 0, 1])
+    array([[ -0.,   0.],
+           [  6.,  24.]])
+    >>> coef = np.arange(4).reshape(2,2) # multidimensional coefficients
+    >>> coef
+    array([[0, 1],
+           [2, 3]])
+    >>> polyvalfromroots([1,2], coef, tensor=True)
+    array([[-1.,  0.],
+           [-0., -1.]])
+    >>> polyvalfromroots([1,2], coef, tensor=False)
+    array([ -1., -1.])
+
+    """
+    r = np.array(r, ndmin=1, copy=0)
+    if r.size == 0:
+        return np.zeros_like(x)
```

Comment:
Yeah, I was debating this when I first wrote it. Since `polyval` is a sum of terms, it makes sense to return the additive identity of `0`. So, we can justify returning the multiplicative identity of `1` since `polyvalfromroots` is a product of terms.